akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/BehaviorSetup.scala

```diff
@@ -173,6 +176,110 @@ private[akka] final class BehaviorSetup[C, E, S](
     }
   }
 
+  // The retention process for SnapshotCountRetentionCriteria looks like this:
+  // 1. Save snapshot after persisting events when shouldSnapshotAfterPersist returned SnapshotWithRetention.
+  // 2. Delete events (when deleteEventsOnSnapshot=true), runs in background.
+  // 3. Delete snapshots (when isOnlyOneSnapshot=false), runs in background.
+
+  def isRetentionInProgress(): Boolean =
+    retentionInProgress
+
+  def retentionProgressSaveSnapshot(sequenceNr: Long): Unit = {
+    retention match {
+      case SnapshotCountRetentionCriteriaImpl(_, _, _) =>
+        internalLogger.debug("Starting retention at seqNr [{}], saving snapshot.", sequenceNr)
+        retentionInProgress = true
+      case _ =>
+    }
+  }
+
+  def retentionProgressSaveSnapshotEnd(sequenceNr: Long, success: Boolean): Unit = {
```

Comment:
same here, recording that it `retentionProgressSaveSnapshotEnded` not ending it?